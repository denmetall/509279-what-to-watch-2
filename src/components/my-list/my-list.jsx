import React from 'react';
import {connect} from 'react-redux';
import Footer from '../footer/footer.jsx';
import {getFavoritesFilms} from "../../selectors";
import withActiveItem from "../../hocs/with-active-item";
import MovieCardsList from "../movie-cards-list/movie-cards-list.jsx";
import Logo from "../logo/logo.jsx";
import UserBlock from "../user-block/user-block.jsx";
import PropTypes from 'prop-types';
import {MovieType} from "../../utils";

const MovieCardsListWrapped = withActiveItem(MovieCardsList);

const MyList = (props) => {
  const {films} = props;

  return (
    <div className="user-page">

      <header className="page-header user-page__head">
        <Logo/>
        <h1 className="page-title user-page__title">My list</h1>
        <UserBlock/>
      </header>

      <section className="catalog">
        <h2 className="catalog__title visually-hidden">Catalog</h2>
        <MovieCardsListWrapped films={films}/>
      </section>
      <Footer />
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    films: getFavoritesFilms(state)
  };
};

MyList.defaultProps = {
  films: []
};

MyList.propTypes = {
  films: PropTypes.arrayOf(MovieType)
};

export {MyList};
export default connect(mapStateToProps)(MyList);


