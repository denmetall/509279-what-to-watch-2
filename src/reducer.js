import filmsMock from './mocks/films';
import {DEFAULT_FILTER} from "./utils";

const initialState = {
  genre: DEFAULT_FILTER,
  films: filmsMock,
  filmsSort: filmsMock,
  promoFilm: filmsMock[0]
};

const ActionType = {
  SET_FILTER_GENRE: `SET_FILTER_GENRE`,
  GET_MOVIES_LIST: `GET_MOVIES_LIST`
};

const getFilterMovies = (moviesGenre) => {
  switch (moviesGenre) {
    case DEFAULT_FILTER:
      return filmsMock;
  }

  return filmsMock.slice().filter((movie) => movie.genre === moviesGenre);
};

const ActionCreator = {
  setGenreFilter: (genre) => ({
    type: ActionType.SET_FILTER_GENRE,
    payload: genre
  }),

  getMoviesList: (genre) => ({
    type: ActionType.GET_MOVIES_LIST,
    payload: getFilterMovies(genre)
  })
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.SET_FILTER_GENRE:
      return Object.assign({}, state, {
        genre: action.payload
      });
    case ActionType.GET_MOVIES_LIST:
      return Object.assign({}, state, {
        filmsSort: action.payload
      });
  }

  return state;
};

export {
  reducer,
  ActionCreator,
  ActionType
};
