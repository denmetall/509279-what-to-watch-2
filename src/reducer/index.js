import {combineReducers} from 'redux';

import {reducer as genre} from './genre/genre';
import {reducer as films} from './films/films';
import {reducer as authorization} from './authorization/authorization';
import {reducer as reviews} from './reviews/reviews';
import {reducer as favorites} from './favorites/favorites';

export default combineReducers(
    {
      genre,
      films,
      authorization,
      reviews,
      favorites
    }
);
