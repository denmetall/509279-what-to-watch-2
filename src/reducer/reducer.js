import {combineReducers} from 'redux';

import {reducer as genre} from './genre/genre';
import {reducer as films} from './films/films';
import {reducer as authorization} from './authorization/authorization';
import {reducer as reviews} from './reviews/reviews';
import {reducer as favorites} from './favorites/favorites';
import {reducer as networkFailed} from './network-failed/network-failed';
import {reducer as loadersReducer} from "react-redux-hoc-loader";

export default combineReducers(
    {
      genre,
      films,
      authorization,
      reviews,
      favorites,
      networkFailed,
      loaders: loadersReducer,
    }
);
