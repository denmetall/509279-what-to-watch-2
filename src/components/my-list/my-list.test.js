import React from 'react';
import {MyList} from './my-list.jsx';
import renderer from 'react-test-renderer';
import filmsMock from '../../mocks/films';
import {BrowserRouter as Router} from 'react-router-dom';
import createNodeMock from '../../mocks/create-node-mock';

import {Provider} from "react-redux";
import {createStore} from "redux";
import reducer from "../../reducer/reducer";

it(`Components MyList renders correctly`, () => {
  const store = createStore(
      reducer,
      window.__REDUX_DEVTOOLS_EXTENSION__ ? window.__REDUX_DEVTOOLS_EXTENSION__() : (f) => f
  );
  const options = {createNodeMock};
  const props = {
    films: filmsMock
  };

  const tree = renderer
    .create(<Provider store={store}><Router><MyList {...props}/></Router></Provider>, options)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
