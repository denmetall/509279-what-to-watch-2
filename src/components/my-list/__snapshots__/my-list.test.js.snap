// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components MyList renders correctly 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link undefined"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <div
      className="user-block"
    >
      <a
        className="user-block__link"
        href="/login"
        onClick={[Function]}
      >
        Sign in
      </a>
    </div>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <video
            height="175"
            muted={true}
            poster="img/johnny-english.jpg"
            width="280"
          >
            <source
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
            />
          </video>
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/1"
            onClick={[Function]}
          >
            Drive
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <video
            height="175"
            muted={true}
            poster="img/macbeth.jpg"
            width="280"
          >
            <source
              src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
            />
          </video>
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/2"
            onClick={[Function]}
          >
            Astral
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <video
            height="175"
            muted={true}
            poster="img/johnny-english.jpg"
            width="280"
          >
            <source
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
            />
          </video>
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/3"
            onClick={[Function]}
          >
            Taxi
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <video
            height="175"
            muted={true}
            poster="img/macbeth.jpg"
            width="280"
          >
            <source
              src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
            />
          </video>
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/4"
            onClick={[Function]}
          >
            Resident evil
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <video
            height="175"
            muted={true}
            poster="img/mindhunter.jpg"
            width="280"
          >
            <source
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
            />
          </video>
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/5"
            onClick={[Function]}
          >
            Brother
          </a>
        </h3>
      </article>
    </div>
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
